src/wasm-as.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2015 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// wasm2asm console tool
+//
+
+#include "support/colors.h"
+#include "support/command-line.h"
+#include "support/file.h"
+#include "wasm-binary.h"
+#include "wasm-s-parser.h"
+
+using namespace cashew;
+using namespace wasm;
+
+int main(int argc, const char *argv[]) {
+  Options options("wasm-as", "Assemble a .wast (WebAssembly text format) into a .wasm (WebAssembly binary format)");
+  options.add("--output", "-o", "Output file (stdout if not specified)",
+              Options::Arguments::One,
+              [](Options *o, const std::string &argument) {
+                o->extra["output"] = argument;
+                Colors::disable();
+              })
+      .add_positional("INFILE", Options::Arguments::One,
+                      [](Options *o, const std::string &argument) {
+                        o->extra["infile"] = argument;
+                      });
+  options.parse(argc, argv);
+
+  std::string input(read_file(options.extra["infile"], options.debug));
+
+  if (options.debug) std::cerr << "s-parsing...\n";
```

Comment:
`std::endl` and `\n` mixture here and below is a bit odd. `std::endl` just flushes buffered output as well (`std::cerr` usually isn't so it's more of a C++ style thing than anything).